### DISCUSSION TITLE
Arcaea Linkplay Webhook for Discord

### DISCUSSION THREAD

**Discussion by burritosoftware** [Lost-MSth/Arcaea-server#135] [Show and tell]
Hey! I wrote a watcher script when running the Link Play server seperately under a systemd service, which will post new rooms and the players in them in Discord as a nice message.

# 📑 [Repository](https://github.com/burritosoftware/arcaea-linkplay-webhook)
![Preview](https://taidums.are-really.cool/5zReRDx.png)

**Comment by Lost-MSth** [comment]
https://github.com/Lost-MSth/Arcaea-server/commit/04010d9c338cf848f734413f017fb81a84536794  I added more logs for Link Play, including the player leaving the room (same with being kicked out), starting playing, players' scores, and some cleaning info.

Well. Because the server cleans the rooms or the players **lazily or periodically**, the log about player leaving room may be late, iff. the player do not use a normal way to leave the room.

I appreciate your work and I consider Link Play module needs more things, such as API to get or control the server, and the log stream via network. These have joined my TODO list. I am open to more advice.

**Reply by burritosoftware** [reply]
That's super cool, thank you! I'll put a note to add support for this later.

I can probably make a way to consider the periodic cleaning, but currently I tie room codes to message IDs and if a new room was made with the same code it will generate a new message ID in its place, which seems to work out. I don't consider it to be too much priority to make sure the status is fully accurate, just long as it's close enough.
